pkg/azureclient/client.go

```diff
@@ -27,30 +40,76 @@ type ResourceSKUsPage interface {
 	Values() []compute.ResourceSku
 }
 
+// ZonePage is a page of results from listing zones.
+type ZonePage interface {
+	NextWithContext(ctx context.Context) error
+	NotDone() bool
+	Values() []dns.Zone
+}
+
+// RecordSetPage is a page of results from listing record sets.
+type RecordSetPage interface {
+	NextWithContext(ctx context.Context) error
+	NotDone() bool
+	Values() []dns.RecordSet
+}
+
 type azureClient struct {
-	config         auth.ClientCredentialsConfig
-	subscriptionID string
+	resourceSKUsClient *compute.ResourceSkusClient
+	recordSetsClient   *dns.RecordSetsClient
+	zonesClient        *dns.ZonesClient
 }
 
 func (c *azureClient) ListResourceSKUs(ctx context.Context) (ResourceSKUsPage, error) {
-	skusClient := compute.NewResourceSkusClient(c.subscriptionID)
-	config := c.config
-	config.Resource = azure.PublicCloud.ResourceManagerEndpoint
-	authorizer, err := config.Authorizer()
+	page, err := c.resourceSKUsClient.List(ctx)
+	return &page, err
+}
+
+func (c *azureClient) CreateOrUpdateZone(ctx context.Context, resourceGroupName string, zone string) (dns.Zone, error) {
+	return c.zonesClient.CreateOrUpdate(ctx, resourceGroupName, zone, dns.Zone{
```

Comment:
It defaults to public if we don't specify anything. I don't see a need to adding it explicitly. Maybe if we add private zone support in the future then it might be useful